DigiIoT.Maui/Platforms/Android/Bluetooth/BeaconAdvertiseService.cs

```diff
@@ -0,0 +1,176 @@
+			_advertiseCallback = new CustomAdvertiseCallback();
+
+			// Update the advertisement name.
+			_bluetoothAdapter.SetName(identifier);
+
+			// Set the advertisement options.
+			var advertiseSettingsBuilder = new AdvertiseSettings.Builder()
+				.SetAdvertiseMode(AdvertiseMode.LowLatency)
+				.SetTxPowerLevel(AdvertiseTx.PowerHigh);
+			if (Build.VERSION.SdkInt >= BuildVersionCodes.UpsideDownCake)
+				advertiseSettingsBuilder = advertiseSettingsBuilder.SetDiscoverable(true);
+			var advertiseSettings = advertiseSettingsBuilder.Build();
+
+			// Generate the advertisement data.
+			var beaconLenght =
+				(payload != null ? payload.Length + companyID.Length + 2 : 0)  // 2 extra bytes for mfg data length and mfg data tag.
+				+ (identifier != null ? identifier.Length + 2 : 0)             // 2 extra bytes for name length and name tag.
+				+ 3;                                                           // 3 extra bytes for flags length, flags tag and flags value.
+			var advertiseDataBuilder = new AdvertiseData.Builder()
+				.SetIncludeDeviceName(identifier != null && beaconLenght <= 31)  // 31 is the maximum allowed beacon length.
+				.SetIncludeTxPowerLevel(false);
+			if (payload != null)
+				advertiseDataBuilder = advertiseDataBuilder.AddManufacturerData(BitConverter.ToUInt16(companyID, 0), payload);
+			var advertiseData = advertiseDataBuilder.Build();
+
+			// Start advertising.
+			_bluetoothLeAdvertiser.StartAdvertising(advertiseSettings, advertiseData, _advertiseCallback);
+		}
+
+		/// <summary>
+		/// Stops advertising.
+		/// </summary>
+		public partial void StopAdvertising()
+		{
+			_bluetoothLeAdvertiser?.StopAdvertising(_advertiseCallback);
+		}
+
+		/// <summary>
+		/// Converts a hexadecimal string to a byte array.
+		/// </summary>
+		/// <param name="hex">The hexadecimal string to convert.</param>
+		/// <returns>A byte array representing the hexadecimal string.</returns>
+		/// <exception cref="ArgumentNullException">Thrown when the input hex
+		/// string is null.</exception>
+		/// <exception cref="System.FormatException">Thrown when the input hex string
+		/// contains non-hexadecimal characters.</exception>
+		public static byte[] HexStringToByteArray(string hex)
+		{
+			if (hex == null)
+				throw new ArgumentNullException(nameof(hex), "Hex string cannot be null.");
+
+			hex = hex.Replace("-", "");
+			if (hex.Length % 2 != 0)
+				hex = "0" + hex;
+
+			byte[] bytes = new byte[hex.Length / 2];
+			for (int i = 0; i < hex.Length; i += 2)
+				bytes[i / 2] = Convert.ToByte(hex.Substring(i, 2), 16);
+			return bytes;
+		}
+	}
+
+	/// <summary>
+	/// Callback required for the advertising process. Notifies about the
+	/// status of the start advertising process.
+	/// </summary>
+	public class CustomAdvertiseCallback : AdvertiseCallback
+	{
+		/// <summary>
+		/// <inheritdoc/>
+		/// </summary>
+		public override void OnStartFailure(AdvertiseFailure errorCode)
+		{
+			base.OnStartFailure(errorCode);
```

Comment:
At this point the upper layer (DigiBLEManager) still thinks the advertisement is running right?